flexmeasures/data/models/time_series.py

```diff
@@ -19,6 +20,93 @@
 class Sensor(db.Model, tb.SensorDBMixin):
     """A sensor measures events. """
 
+    def search_beliefs(
+        self,
+        event_time_window: Tuple[Optional[datetime_type], Optional[datetime_type]] = (
+            None,
+            None,
+        ),
+        belief_time_window: Tuple[Optional[datetime_type], Optional[datetime_type]] = (
+            None,
+            None,
+        ),
+        source: Optional[Union[int, List[int], str, List[str]]] = None,
+    ):
+        """Search all beliefs about events for this sensor.
+
+        :param event_time_window: search only events within this time window
+        :param belief_time_window: search only beliefs within this time window
+        :param source: search only beliefs by this source (pass its name or id) or list of sources"""
+        return TimedBelief.search_all(
+            sensor=self,
+            event_time_window=event_time_window,
+            belief_time_window=belief_time_window,
+            source=source,
+        )
+
+    def __repr__(self) -> str:
+        return f"<Sensor {self.id}: {self.name}>"
+
+
+class TimedBelief(db.Model, tb.TimedBeliefDBMixin):
+    """A timed belief holds a precisely timed record of a belief about an event."""
+
+    @declared_attr
+    def source_id(cls):
+        return db.Column(db.Integer, db.ForeignKey("data_source.id"), primary_key=True)
+
+    sensor = db.relationship("Sensor", backref=db.backref("beliefs", lazy=True))
+    source = db.relationship("DataSource", backref=db.backref("beliefs", lazy=True))
+
+    @classmethod
+    def search_all(
```

Comment:
Maybe this can simply "search"? And we could use that term also in `tb.TimedBeliefDBMixin` actually.